external-plugins/rehearse/plugin/handler/handler.go

```diff
@@ -0,0 +1,504 @@
+			log.WithError(err).Infoln("Could not write temp file")
+			return nil, err
+		}
+		pc, err := config.Load(prowConfigTmp, jobConfigTmp)
+		if err != nil {
+			log.WithError(err).Errorf("Could not load job config from path %s at git ref %s", jobConfigTmp, ref)
+			return nil, err
+		}
+		configs[changedJobConfig] = pc
+	}
+
+	return configs, nil
+}
+
+// modifiedJobConfigs generates an array of absolute paths for modified job configs
+func (h *gitHubEventsHandler) modifiedJobConfigs(changedFiles []string) ([]string, error) {
+	var absModifiedProwConfigs []string
+	for _, changedFile := range changedFiles {
+		if strings.HasPrefix(changedFile, h.jobsConfigBase) {
+			if strings.HasSuffix(changedFile, ".yaml") || strings.HasSuffix(changedFile, ".yml") {
+				log.Infof("A modified config found: %s", changedFile)
+				absModifiedProwConfigs = append(absModifiedProwConfigs, changedFile)
+			}
+		}
+		log.Debugf("Skipping file: %s. Not a Prow/Jobs config", changedFile)
+	}
+	return absModifiedProwConfigs, nil
+}
+
+func jobKeyFunc(repo string, presubmit config.JobBase) string {
+	return fmt.Sprintf("%s#%s", repo, presubmit.Name)
+}
+
+func repoFromJobKey(jobKey string) string {
+	s := strings.Split(jobKey, "#")
+	r := s[:1]
+	return strings.Join(r, "/")
+}
+
+func hashPeriodicsConfig(periodics []config.Periodic) map[string]config.Periodic {
+	p := map[string]config.Periodic{}
+	for _, periodic := range periodics {
+		p[periodic.JobBase.Name] = periodic
+	}
+	return p
+}
+
+func hashPresubmitsConfig(presubmits map[string][]config.Presubmit) map[string]config.Presubmit {
+	presubmitsFlat := map[string]config.Presubmit{}
+	for repo, presubmitsForRepo := range presubmits {
+		for _, presubmit := range presubmitsForRepo {
+			presubmitsFlat[jobKeyFunc(repo, presubmit.JobBase)] = presubmit
+		}
+	}
+	return presubmitsFlat
+}
+
+// catFile executes a git cat-file command in the specified git dir and returns bytes representation of the file
+func catFile(log *logrus.Logger, gitDir, file, refspec string) ([]byte, int) {
+	cmd := exec.Command("git", "-C", gitDir, "cat-file", "-p", fmt.Sprintf("%s:%s", refspec, file))
+	log.Debugf("Executing git command: %+v", cmd.Args)
+	out, _ := cmd.CombinedOutput()
+	return out, cmd.ProcessState.ExitCode()
+}
+
+func writeTempFile(log *logrus.Logger, basedir string, content []byte) (string, error) {
+	tmpfile, err := ioutil.TempFile(basedir, "job-config")
+	if err != nil {
+		log.WithError(err).Errorf("Could not create temp file for job config.")
+		return "", err
+	}
+	defer tmpfile.Close()
+	_, err = tmpfile.Write(content)
+	if err != nil {
+		log.WithError(err).Errorf("Could not write data to file: %s", tmpfile.Name())
+		return "", err
+	}
+	tmpfile.Sync()
+	return tmpfile.Name(), nil
+}
+
+func addExtraRefs(refs []prowapi.Refs, org, repo, ref string) []prowapi.Refs {
+	// We don't want to modify the refs on the original object, so we make our copy of it
+	r := reflect.ValueOf(make([]prowapi.Refs, len(refs)))
```

Comment:
Why do you not just add it directly? If you don't want to add it directly, consider something like this:

```go
func generateTargetRepoRef (job v1.ProwJob, org, repo, ref string) prowapi.Refs {
    return prowapi.Refs{
prowapi.Refs{
				Repo:    repo,
				Org:     org,
				WorkDir: !workdirAlreadyDefined(refs),
				BaseRef: ref,
}
}
```

and then

```
job.Refs  = append(job.Refs, generateTargetRepoRef(job, org, repo, ref))
```